### PR TITLE
Extend ingress e2e

### PR DESCRIPTION
**Pull request by bprashanth**
Splits the test into a cross platform conformance list, and platform specific bits that exercise features through annotations. Also exercises the features in https://github.com/kubernetes/contrib/pull/1133. Assigning to Girish, simply because I assigned the other pr to Minhan.

### CODE REVIEW

File: test/e2e/ingress.go [hunk 1]
```diff
@@ -19,188 +19,427 @@ package e2e
...
+	"k8s.io/kubernetes/pkg/util/sets"
 	"k8s.io/kubernetes/pkg/util/wait"
+	utilyaml "k8s.io/kubernetes/pkg/util/yaml"
 	"k8s.io/kubernetes/test/e2e/framework"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
-// Before enabling this test you must make sure the associated project has
```
Comment (girishkalele) [inline test/e2e/ingress.go]: This quota warning seems like a good thing to keep, unless its no longer applicable ?

Comment (bprashanth) [inline test/e2e/ingress.go]: I just moved it to the gce specific segment https://github.com/kubernetes/kubernetes/pull/26821/files/da2351fe26ec9d3c18a79db5bf724df05c7113e2#diff-ec4b96325ef379121837e76bb579c833R166

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #24476

**Comment by bprashanth** [comment]
@girishkalele if you have time today, would be good to get this test to run over the weekend. https://github.com/kubernetes/contrib/pull/1133 merged, I want to get green, bump the image, still see green.

File: test/e2e/ingress.go [hunk 2]
```diff
@@ -19,188 +19,427 @@ package e2e
...
+				t.execute()
+				By(t.exitLog)
+				jig.waitForIngress()
+			}
+		})
 
-	// GCE only allows names < 64 characters, and the loadbalancer controller inserts
-	// a single character of padding.
-	nameLenLimit = 62
+		It("shoud crate ingress with given static-ip ", func() {
```
Comment (girishkalele) [inline test/e2e/ingress.go]: typo - "should create"

File: test/e2e/ingress.go [hunk 3]
```diff
@@ -226,26 +465,30 @@ func gcloudUnmarshal(resource, regex, project string, out interface{}) {
...
 }
 
-func gcloudDelete(resource, name, project string) {
+func gcloudDelete(resource, name, project string, args ...string) error {
 	framework.Logf("Deleting %v: %v", resource, name)
-	output, err := exec.Command("gcloud", "compute", resource, "delete",
-		name, fmt.Sprintf("--project=%v", project), "-q").CombinedOutput()
+	argList := append([]string{"compute", resource, "delete", name, fmt.Sprintf("--project=%v", project), "-q"}, args...)
+	output, err := exec.Command("gcloud", argList...).CombinedOutput()
 	if err != nil {
```
Comment (girishkalele) [inline test/e2e/ingress.go]: Are gcloud commands flaky ? Retries useful ?

File: test/e2e/ingress.go [hunk 4]
```diff
@@ -269,7 +512,25 @@ func (cont *IngressController) init() {
 	}
 }
 
-func (cont *IngressController) Cleanup(del bool) error {
+func (cont *GCEIngressController) staticIP(name string) string {
+	output, err := exec.Command("gcloud", "compute", "addresses", "create", name, "--global").CombinedOutput()
```
Comment (girishkalele) [inline test/e2e/ingress.go]: Are gcloud commands flaky ? Retries useful ?

File: test/e2e/ingress.go [hunk 5]
```diff
@@ -281,12 +542,7 @@ func (cont *IngressController) Cleanup(del bool) error {
 			for _, f := range fwList {
 				msg += fmt.Sprintf("%v\n", f.Name)
 				if del {
-					framework.Logf("Deleting forwarding-rule: %v", f.Name)
-					output, err := exec.Command("gcloud", "compute", "forwarding-rules", "delete",
-						f.Name, fmt.Sprintf("--project=%v", cont.Project), "-q", "--global").CombinedOutput()
-					if err != nil {
-						framework.Logf("Error deleting forwarding rules, output: %v\nerror:%v", string(output), err)
-					}
+					gcloudDelete("forwarding-rules", f.Name, cont.Project, "--global")
```
Comment (girishkalele) [inline test/e2e/ingress.go]: Would parallelization help when deleting all these gcloud resources ?

File: test/e2e/testing-manifests/ingress/http/svc.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+spec:
+  type: NodePort
+  ports:
+  - port: 80
+    targetPort: 8080
+    protocol: TCP
+    name: http
+  selector:
+    app: echoheaders
+---
```
Comment (girishkalele) [inline test/e2e/testing-manifests/ingress/http/svc.yaml]: Didn't know you could do that with '---' in a yaml file.

**Comment by girishkalele** [comment]
@bprashanth 

Don't need to implement the retry and parallelization suggestions, maybe in a followup.
One typo that needs fixing, rest looks good to me.

File: test/e2e/ingress.go (hunk 3 above)
Comment (bprashanth) [inline test/e2e/ingress.go]: yeah we retry the overall cleanup call https://github.com/kubernetes/kubernetes/pull/26821/files#diff-ec4b96325ef379121837e76bb579c833R198

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #IGNORE (no logs of any sort, assuming infra flake)

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #IGNORE (merge queue was broken on friday evening)

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #27284

**Comment by bprashanth** [comment]
@k8s-bot unit test this github issue: #27285

**Comment by bprashanth** [comment]
Last failure is https://github.com/kubernetes/kubernetes/issues/27287, I'm going to let it sit for a while so we don't continue hammering out qps limits

**Comment by bprashanth** [comment]
@k8s-bot test this github issue: #27287

**Comment by bprashanth** [comment]
Making this p1 to fix https://github.com/kubernetes/kubernetes/issues/27486

**Pull request merged**